src/Control/HTTPRequest.php

```diff
@@ -579,7 +579,7 @@ public function match($pattern, $shiftOnSuccess = false)
                         $shiftCount = sizeof($patternParts ?? []);
                         $remaining = count($this->dirParts ?? []) - $i;
                         for ($j = 1; $j <= $remaining; $j++) {
-                            $arguments["$${j}"] = $this->dirParts[$j + $i - 1];
+                            $arguments['$' . $j] = $this->dirParts[$j + $i - 1];
```

Comment:
Have confirmed with local testing that this is the correct way to de-deprecate this extremely confusing syntax